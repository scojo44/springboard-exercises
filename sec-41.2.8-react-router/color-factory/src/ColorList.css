.ColorList {
  margin: 1rem;
}