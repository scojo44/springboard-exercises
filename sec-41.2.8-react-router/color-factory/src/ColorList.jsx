import React from "react"
import {Link} from 'react-router-dom'

const ColorList = ({colors}) => {
  return (
    <div className="ColorList">
      <p><Link to="new">Add a new color</Link></p>
      <menu>
        {colors.map(c => <li key={c.id}><Link to={c.name}>{c.name}</Link></li>)}
      </menu>
    </div>
  );
};

export default ColorList
