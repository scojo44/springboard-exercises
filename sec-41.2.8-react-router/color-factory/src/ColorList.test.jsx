import {render} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import ColorList from './ColorList'

const TEST_COLORS = [
  {
    id: 1,
    name: 'purple',
    value: '#FF00FF'
  }
]

describe('ColorList Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <ColorList colors={TEST_COLORS} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <ColorList colors={TEST_COLORS} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
