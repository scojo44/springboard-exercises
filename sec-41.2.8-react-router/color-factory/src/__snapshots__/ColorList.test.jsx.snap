// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColorList Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColorList"
  >
    <p>
      <a
        href="/new"
      >
        Add a new color
      </a>
    </p>
    <menu>
      <li>
        <a
          href="/purple"
        >
          purple
        </a>
      </li>
    </menu>
  </div>
</DocumentFragment>
`;
